diff --git a/python/sglang/srt/layers/attention/dsa_backend.py b/python/sglang/srt/layers/attention/dsa_backend.py
index 13f78f2532..6c57176110 100644
--- a/python/sglang/srt/layers/attention/dsa_backend.py
+++ b/python/sglang/srt/layers/attention/dsa_backend.py
@@ -383,7 +383,7 @@ class DeepseekSparseAttnBackend(
                 16 // self.num_q_heads if self.num_q_heads < 16 else 1
             )
             self.num_head_padded = self.num_q_heads * self.head_repeat_factor
-            self.aiter_dsa_max_split_per_batch = 64
+            self.aiter_dsa_max_split_per_batch = 16
             self.aiter_dsa_metadata_capacity = 0
             self.aiter_dsa_metadata_max_seqlen_q = 0
             self.aiter_dsa_metadata_q_dtype = None
@@ -449,7 +449,7 @@ class DeepseekSparseAttnBackend(
             is_sparse=True,
             fast_mode=False,
             num_kv_splits=self.aiter_dsa_max_split_per_batch,
-            intra_batch_mode=True,
+            intra_batch_mode=False,
         )
 
         return (
@@ -546,9 +546,8 @@ class DeepseekSparseAttnBackend(
             max_seqlen_qo=max_seqlen_q,
             uni_seqlen_qo=max_seqlen_q,
             fast_mode=False,
-            topk=self.dsa_index_topk,
             max_split_per_batch=self.aiter_dsa_max_split_per_batch,
-            intra_batch_mode=True,
+            intra_batch_mode=False,
             dtype_q=q_dtype,
             dtype_kv=kv_dtype,
         )
@@ -561,7 +560,7 @@ class DeepseekSparseAttnBackend(
             "reduce_indptr": self.aiter_dsa_reduce_indptr,
             "reduce_final_map": self.aiter_dsa_reduce_final_map,
             "reduce_partial_map": self.aiter_dsa_reduce_partial_map,
-            "intra_batch_mode": True,
+            "intra_batch_mode": False,
             "num_kv_splits": self.aiter_dsa_max_split_per_batch,
         }
 
@@ -2180,9 +2179,9 @@ class DeepseekSparseAttnBackend(
         q = q_all.reshape(-1, layer.tp_q_head_num * layer.head_dim)
 
         if layer.head_dim != layer.v_head_dim:
-            o = q.new_empty((q.shape[0], layer.tp_q_head_num * layer.v_head_dim))
+            o = q.new_empty((q.shape[0], layer.tp_q_head_num * layer.v_head_dim), dtype=torch.bfloat16)
         else:
-            o = torch.empty_like(q)
+            o = torch.empty_like(q, dtype=torch.bfloat16)
 
         if self.need_pad_heads:
             q_kernel = q.view(
@@ -2193,7 +2192,8 @@ class DeepseekSparseAttnBackend(
                     q.shape[0],
                     layer.tp_q_head_num * self.head_repeat_factor,
                     layer.v_head_dim,
-                )
+                ),
+                dtype=torch.bfloat16,
             )
         else:
             q_kernel = q.view(-1, layer.tp_q_head_num, layer.head_dim)
@@ -2204,6 +2204,8 @@ class DeepseekSparseAttnBackend(
         aiter_persistent_kwargs = {}
         if kv_cache.dtype == fp8_dtype:
             kv_scale = torch.ones((), dtype=torch.float32, device=q_kernel.device)
+        if q_kernel.dtype == fp8_dtype:
+            q_scale = torch.ones((), dtype=torch.float32, device=q_kernel.device)
 
         kv_indptr = self.kv_indptr
 
@@ -2258,9 +2260,9 @@ class DeepseekSparseAttnBackend(
         q = q_all.reshape(-1, layer.tp_q_head_num * layer.head_dim)
 
         if layer.head_dim != layer.v_head_dim:
-            o = q.new_empty((num_tokens, layer.tp_q_head_num * layer.v_head_dim))
+            o = q.new_empty((num_tokens, layer.tp_q_head_num * layer.v_head_dim), dtype=torch.bfloat16)
         else:
-            o = torch.empty_like(q)
+            o = torch.empty_like(q, dtype=torch.bfloat16)
 
         if self.need_pad_heads:
             q_kernel = q.view(
@@ -2271,7 +2273,8 @@ class DeepseekSparseAttnBackend(
                     num_tokens,
                     layer.tp_q_head_num * self.head_repeat_factor,
                     layer.v_head_dim,
-                )
+                ),
+                dtype=torch.bfloat16,
             )
         else:
             q_kernel = q.view(-1, layer.tp_q_head_num, layer.head_dim)
@@ -2282,6 +2285,8 @@ class DeepseekSparseAttnBackend(
         aiter_persistent_kwargs = {}
         if kv_cache.dtype == fp8_dtype:
             kv_scale = torch.ones((), dtype=torch.float32, device=q_kernel.device)
+        if q_kernel.dtype == fp8_dtype:
+            q_scale = torch.ones((), dtype=torch.float32, device=q_kernel.device)
 
         non_minus1_mask = page_table_1 != -1
         non_minus1_counts = non_minus1_mask.sum(dim=1)
